src/main/java/ognl/OgnlRuntime.java

```diff
@@ -3963,19 +3966,22 @@ void put(Class clazz, String propertyName, Method method)
             if (methodsByPropertyName == null)
             {
                 methodsByPropertyName = new ConcurrentHashMap<String, Method>();
-                this.cache.put(clazz, methodsByPropertyName);
+                ConcurrentHashMap<String,Method> old = this.cache.putIfAbsent(clazz, methodsByPropertyName);
+                if (null != old) {
+                    methodsByPropertyName = old;
+                }
             }
-            methodsByPropertyName.put(propertyName, (method == null? NULL_REPLACEMENT : method));
+            methodsByPropertyName.putIfAbsent(propertyName, (method == null? NULL_REPLACEMENT : method));
         }
 
-        boolean containsKey(Class clazz, String propertyName)
-        {
-            ConcurrentHashMap<String,Method> methodsByPropertyName = this.cache.get(clazz);
-            if (methodsByPropertyName == null)
-                return false;
-
-            return methodsByPropertyName.containsKey(propertyName);
-        }
+//        boolean containsKey(Class clazz, String propertyName)
+//        {
+//            ConcurrentHashMap<String,Method> methodsByPropertyName = this.cache.get(clazz);
+//            if (methodsByPropertyName == null)
+//                return false;
+//
+//            return methodsByPropertyName.containsKey(propertyName);
+//        }
```

Comment:
Please remove the code instead of commenting it out

Reply:
done